lib/operdebug.g

```diff
@@ -0,0 +1,69 @@
+##############################################################################
+##  operdebug.g                                                   Frank Lübeck
+##  
+##  Non-documented  utilities to  write out  infos  about all  methods of  all
+##  operations with  location and rank, and  to write out all  filters by name
+##  and their rank. These can be useful to debug larger changes that influence
+##  the ranking of filters and methods.
+##  
+BIND_GLOBAL("WriteMethodOverview", function(fname)
+  local name, sline, f;
+  name := function(f)
+    local path, ls, s;
+    path := FilenameFunc(f);
+    if path=fail then
+      return NameFunction(f);
+    fi;
+    for s in GAPInfo.RootPaths do
+      ls := Length(s);
+      if Length(path) >= ls and path{[1..ls]}=s then
+        return path{[ls+1..Length(path)]};
+      fi;
+    od;
+    return path;
+  end;
+  sline := function(f)
+    local n;
+    n := StartlineFunc(f);
+    if n=fail then
+      return "\c";
+    fi;
+    return Concatenation(":",String(n),"\c");
+  end;
+  f := function()
+    local nam, m, s, op, i, j;
+    for op in OPERATIONS do
+      nam := NameFunction(op);
+      for i in [0..6] do
+        m := METHODS_OPERATION(op, i);
+        s := 4+i;
```

Comment:
This needs to be changed to `s := BASE_SIZE_METHODS_OPER_ENTRY + i;`

Reply:
I also suggest rebasing this PR on master and verifying these functions still work. Ideally, there'd be a test case for them, too, just to make sure we don't break them horribly. Of course one may argue they are just debugging tools; but also debugging tools are more useful if in order to use them you don't first have to track down what caused a regression in them some time in the past ;-).